public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -106,6 +107,47 @@ async function fetchDashboard(
   }
 }
 
+const trackQueriesOnInitDashboard = (
+  panels: PanelModel[],
+  dashboardId: string,
+  orgId?: number,
+  userId?: number,
+  grafanaVersion?: string
+): void => {
+  const queries = getQueriesFromDashboardPerDS(panels);
+  const datasourceSrv = getDataSourceSrv();
+
+  Object.keys(queries).forEach((datasourceId) => {
+    datasourceSrv.get(datasourceId).then((ds) => {
+      if (ds.trackQueries) {
+        ds.trackQueries(queries[datasourceId], dashboardId, orgId, userId, grafanaVersion);
+      }
+    });
+  });
+};
+
+const getQueriesFromDashboardPerDS = (
```

Comment:
**Question:** as this is more concerned about working with "panels", shouldn't we call it something like `getQueriesForPanels()` or `getQueriesForPanelsPerDS()` instead?